app/controllers/spree/taxons_controller_decorator.rb

```diff
@@ -9,24 +10,22 @@ def show
       end
 
       @taxon = Spree::Taxon.friendly.find(params[:id])
-      
       return unless @taxon
+      @searcher = build_searcher(params.merge(taxon: @taxon, include_images: true))
+        list = ['descend_by_popularity', 'ascend_by_name', 'descend_by_name']
+        if list.include?(params[:sorting])
+            sorting_scope = params[:sorting].try(:to_sym) || :ascend_by_name
+        else
+            sorting_scope = :ascend_by_name
+        end
+      if params[:sorting] == "bestsellers"
 
-      search_params = params.merge(
-        current_store: current_store,
-        taxon: @taxon,
-        include_images: true
-      )
-      @searcher = build_searcher(search_params)
-      list      = ['descend_by_popularity', 'ascend_by_name', 'descend_by_name']
-
-      if list.include?(params[:sorting])
-        sorting_scope = params[:sorting].try(:to_sym) || :ascend_by_name
-      else
-        sorting_scope = :ascend_by_name
-      end 
-      @products = @searcher.retrieve_products.send(sorting_scope).page(params[:page]).per(params['per_page'])
-      @taxonomies = Spree::Taxonomy.includes(root: :children).where('name ILIKE ?','%PRODUCTS%')
+        @products = @searcher.retrieve_products.send(sorting_scope).best_sellers.page(params[:page]).per(params[:per_page])
+      else  
+        @products = @searcher.retrieve_products.send(sorting_scope).page(params[:page]).per(params[:per_page])
+      end
+      @taxonomies = Spree::Taxonomy.includes(root: :children).where('name ILIKE ?',"%Category%")
```

Comment:
We dnt need to hardcoded the taxonomy here. We should as about this from client for natures. Always discuss these type of questions with me.